flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/sink/writer/AsyncSinkWriter.java

```diff
@@ -310,8 +309,32 @@ private void flush() {
 
         inFlightRequestsCount++;
         submitRequestEntries(batch, requestResult);
-        numRecordsOutCounter.inc(batchSize);
+    }
+
+    /**
+     * Creates the next batch of request entries while respecting the {@code maxBatchSize} and
+     * {@code flushOnBufferSizeBytes}. Also adds these to the metrics counters.
+     */
+    private List<RequestEntryT> createNextAvailableBatch() {
+        int batchSize = Math.min(maxBatchSize, bufferedRequestEntries.size());
+        List<RequestEntryT> batch = new ArrayList<>(batchSize);
+
+        int batchSizeBytes = 0;
+        for (int i = 0; i < batchSize; i++) {
+            if (batchSizeBytes + bufferedRequestEntries.peek().getSize()
+                    >= maxBatchSizeInBytes) {
+                break;
+            }
```

Comment:
It is ok if it is equal to maxBatchSizeInBytes? I think you just need `>` comparison